docs: add `add-on` docs (#272)
sveltejs/cli: documentation/docs/30-add-ons/05-drizzle.md

---
title: drizzle
---

## Usage

```bash
npx sv add drizzle
```

## Description

> [Drizzle ORM](https://orm.drizzle.team/) is a headless TypeScript ORM with a head.

Reviewer: headless with a head? lol. I just checked and that's their tagline, but it's completely unintelligible, so I think we should write our own description for it :smile:

Some short description about what drizzle is and what we set up

## Options

### database

Which database variant to use:

- `postgresql` — tbd
- `mysql` — tbd
- `sqlite` — runs everywhere, local database

```bash
npx sv add --drizzle=postgresql
```

### client

The sql client to use, depends on `database`:

- For `postgresql`: `postgres.js`, `neon`,
- For `mysql`: `mysql2`, `planetscale`
- For `sqlite`: `better-sqlite3`, `libsql`, `turso`

```bash
npx sv add --drizzle=postgresql,postgres.js
```

> 

Reviewer: ```suggestion

Drizzle is compatible with well over a dozen database drivers. We just offer a few of the most common ones here for simplicity, but if you'd like to use another one you can choose one as a placeholder and swap it out for another after setup by choosing from [Drizzle's full list of compatible drivers](https://orm.drizzle.team/docs/connect-overview#next-steps).

```

---
title: drizzle
---

## Usage

```bash
npx sv add drizzle
```

## Description

[Drizzle ORM](https://orm.drizzle.team/) is a headless TypeScript ORM with a head.
Some short description about what drizzle is and what we set up

## Options

### database

Which database variant to use:

> - `postgresql` — tbd
> - `mysql` — tbd
> - `sqlite` — runs everywhere, local database

Reviewer: ```suggestion
- `postgresql` — the most popular open source database
- `mysql` — another popular open source database
- `sqlite` — file-based database not requiring a database server
```

```bash
npx sv add --drizzle=postgresql
```

### client

> The sql client to use, depends on `database`:

Reviewer: ```suggestion
The SQL client to use, depends on `database`:
```

- For `postgresql`: `postgres.js`, `neon`,
- For `mysql`: `mysql2`, `planetscale`
- For `sqlite`: `better-sqlite3`, `libsql`, `turso`

```bash
npx sv add --drizzle=postgresql,postgres.js
```

Drizzle is compatible with well over a dozen database drivers. We just offer a few of the most common ones here for simplicity, but if you'd like to use another one you can choose one as a placeholder and swap it out for another after setup by choosing from [Drizzle's full list of compatible drivers](https://orm.drizzle.team/docs/connect-overview#next-steps).

### docker

> Whether to add docker compose configuration. Only available for [`database`](#Options-database) `postgresql` or `mysql`

Reviewer: ```suggestion
Whether to add Docker Compose configuration. Only available for [`database`](#Options-database) `postgresql` or `mysql`
```
